cpp/include/cudf_test/default_stream.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf {
+namespace test {
+
+/**
+ * @brief Get the default stream to use for tests.
+ *
+ * @return The default stream to use for tests.
+ */
+rmm::cuda_stream_view const get_default_stream();
```

Comment:
Sorry I don't get this. Why do we need this alias function to `cudf::get_default_stream()`? If we indeed need it, please add more comment to clarify.

Reply:
No worries, this is a good question and the answer is not obvious. The purpose of this function is to provide a symbol that may be overloaded with a preloaded library, which is what all the stream verification is designed to leverage. The idea is that if all tests pass `cudf::test::get_default_stream` as an argument to functions that accept a stream, we can overload the behavior of the function at runtime to provide a specific stream and then track all CUDA API calls to ensure that everything runs on that new stream instead of on e.g. the CUDA default stream (which would indicate that somewhere we are failing to pass the stream all the way down the call stack).